RodSoriano/nekupu
# 🚀  Finished deployment setup

## Description

This PR refines the deployment process, introduces code refactoring, and adds `.env` support to the frontend to securely handle sensitive keys. Key updates include:

- Removal of unnecessary `console.log` statements.
- Added `GET` endpoint for fetching categories in the backend.
- Enhanced error handling with a custom exception for the `uploadFile` method in `S3Service`.
- Set a default `userId` for the `create` method in `PostService`.
- Implemented `.env` support in the frontend to securely manage environment-specific configurations.

## Type of Change

Please indicate the type of change included in this PR:


## How to Test

1. **Set up environment variables**: Duplicate the `.env.example` file and fill in the required values in the `.env` file for both the backend and frontend.
2. **Build and start**: Run `npm run build` from the root, then `npm run start`. Ensure that all services start without issues and verify that environment variables are correctly configured for your local setup.

## Verification Checklist

- [x] Code quality and structure adhere to best practices.
- [x] No significant performance regressions.
- [x] Security considerations are applied, with no new vulnerabilities introduced.
- [x] Documentation is current and reflects the latest changes.
- [ ] Test coverage remains at acceptable levels or has been improved.

## Suggested Reviewers

- @RodSoriano

## Labels

- `enhancement`
